From 7745df73f1e015369e8b92ab9605c87cc26827a4 Mon Sep 17 00:00:00 2001
From: Wanja Schulze <rajaschu@googlemail.com>
Date: Wed, 4 Apr 2018 00:46:44 +0200
Subject: [PATCH] Switch to a secure gps connection

Because the XTRA and XTRA2 data files are served over HTTP without SSL,
this allows an attacker to mount a MITM attack on the network level and
modify the GPS assistance data while in transit.

Informations and fix taken from:
https://wwws.nightwatchcybersecurity.com/2016/12/05/cve-2016-5341/

Change-Id: I4b2c16acc7316c56d6d50d824bf951b248df16e0
---
 gps/gps.conf_AS      | 6 +++---
 gps/gps.conf_AS_SUPL | 6 +++---
 gps/gps.conf_EU      | 6 +++---
 gps/gps.conf_EU_SUPL | 6 +++---
 gps/gps.conf_US      | 6 +++---
 gps/gps.conf_US_SUPL | 6 +++---
 6 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/gps/gps.conf_AS b/gps/gps.conf_AS
index ce3a4e2..052aca0 100644
--- a/gps/gps.conf_AS
+++ b/gps/gps.conf_AS
@@ -1,3 +1,3 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
diff --git a/gps/gps.conf_AS_SUPL b/gps/gps.conf_AS_SUPL
index a0e2a7d..ddd2be3 100644
--- a/gps/gps.conf_AS_SUPL
+++ b/gps/gps.conf_AS_SUPL
@@ -1,5 +1,5 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
 SUPL_HOST=supl.google.com
 SUPL_PORT=7276
diff --git a/gps/gps.conf_EU b/gps/gps.conf_EU
index ce3a4e2..052aca0 100644
--- a/gps/gps.conf_EU
+++ b/gps/gps.conf_EU
@@ -1,3 +1,3 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
diff --git a/gps/gps.conf_EU_SUPL b/gps/gps.conf_EU_SUPL
index a0e2a7d..ddd2be3 100644
--- a/gps/gps.conf_EU_SUPL
+++ b/gps/gps.conf_EU_SUPL
@@ -1,5 +1,5 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
 SUPL_HOST=supl.google.com
 SUPL_PORT=7276
diff --git a/gps/gps.conf_US b/gps/gps.conf_US
index ce3a4e2..052aca0 100644
--- a/gps/gps.conf_US
+++ b/gps/gps.conf_US
@@ -1,3 +1,3 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
diff --git a/gps/gps.conf_US_SUPL b/gps/gps.conf_US_SUPL
index a0e2a7d..ddd2be3 100644
--- a/gps/gps.conf_US_SUPL
+++ b/gps/gps.conf_US_SUPL
@@ -1,5 +1,5 @@
-XTRA_SERVER_1=http://xtra1.gpsonextra.net/xtra.bin
-XTRA_SERVER_2=http://xtra2.gpsonextra.net/xtra.bin
-XTRA_SERVER_3=http://xtra3.gpsonextra.net/xtra.bin
+XTRA_SERVER_1=https://xtrapath1.izatcloud.net/xtra.bin
+XTRA_SERVER_2=https://xtrapath2.izatcloud.net/xtra.bin
+XTRA_SERVER_3=https://xtrapath3.izatcloud.net/xtra.bin
 SUPL_HOST=supl.google.com
 SUPL_PORT=7276
-- 
2.7.4

